src/source/PeerConnection/SessionDescription.c

```diff
@@ -316,6 +326,64 @@ PCHAR fmtpForPayloadType(UINT64 payloadType, PSessionDescription pSessionDescrip
     return NULL;
 }
 
+/*
+ * Extracts a (hex) value after the provided prefix string. Returns true if
+ * successful.
+ */
+static BOOL readHexValue(const PCHAR input, const PCHAR prefix, PINT32 value) {
+  const PCHAR substr = STRSTR(input, prefix);
+  if (substr != NULL) {
+    if (sscanf(substr + STRLEN(prefix), "%x", value) == 1) {
+      return TRUE;
+    }
+  }
+  return FALSE;
+}
+
+/*
+ * Scores the provided fmtp string based on this library's ability to
+ * process various types of H264 streams. A score of 0 indicates an
+ * incompatible fmtp line. Beyond this, a higher score indicates more
+ * compatibility with the desired characteristics, packetization-mode=1,
+ * level-asymmetry-allowed=1, and inbound match with our preferred
+ * profile-level-id.
+ */
+INT32 getH264FmtpScore(PCHAR fmtp) {
+  INT32 profileId = 0, packetizationMode = 0, levelAsymmetry = 0;
```

Comment:
UINT32

Reply:
Note: If using sscanf(str, "%x", ...), I think these have to be INT32s... see "type of argument" for x, X: https://www.tutorialspoint.com/c_standard_library/c_function_sscanf.htm